### PR TITLE
[ONNX] Support aten::atan2 in torchscript exporter

### LINKED ISSUE #51334: ONNX export of atan2
## 🐛 Bug
Try converting torch model to onnx, atan2 is not supported.

RuntimeError: Exporting the operator atan2 to ONNX opset version 9 is not supported. Please open a bug to request ONNX export support for the missing operator.

## To Reproduce
import torch
import torch.nn as nn
import torch.nn.functional as F
from torch.autograd import Variable
import numpy as np

class Net(nn.Module):
   def __init__(self):
       super(Net, self).__init__()
       self.layer = torch.nn.Linear(1, 1)

   def forward(self, x):
       x = self.layer(x)
       x = torch.atan2(x, torch.tensor(1.0))   
       return x
net = Net()
print(net)

x = np.random.rand(100)
y = np.sin(x) * np.power(x,3) + 2*x + np.random.rand(100)*0.8

x = torch.from_numpy(x.reshape(-1,1)).float()
y = torch.from_numpy(y.reshape(-1,1)).float()

optimizer = torch.optim.SGD(net.parameters(), lr=0.1)
loss_func = torch.nn.MSELoss()
inputs = Variable(x)
outputs = Variable(y)
for i in range(100):
   prediction = net(inputs)
   loss = loss_func(prediction, outputs) 
   optimizer.zero_grad()
   loss.backward()
   optimizer.step()
   print(i, loss.detach().numpy())

inputs = torch.randn([2, 1]) 
input_names = ["input"]
output_names = ["output"]
onnx_filename = "./model_regression.onnx"
torch.onnx.export(net, inputs, onnx_filename,
	verbose=True, input_names=input_names, output_names=output_names,
	opset_version=9)`

## Expected behavior

Want to convert torch model to onnx then to tflite.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)

[pip3] numpy==1.19.4
[pip3] torch==1.7.0
[pip3] torchaudio==0.7.0a0+ac17b64
[pip3] torchaudio-contrib==0.1
[pip3] torchcontrib==0.0.2
[pip3] torchvision==0.8.1
[conda] blas                      1.0                         mkl    defaults
[conda] cudatoolkit               11.0.221             h6bb024c_0    defaults
[conda] mkl                       2020.2                      256    defaults
[conda] mkl-service               2.3.0            py38he904b0f_0    defaults
[conda] mkl_fft                   1.2.0            py38h23d657b_0    defaults
[conda] mkl_random                1.1.1            py38h0573a6f_0    defaults
[conda] numpy                     1.19.4                   pypi_0    pypi
[conda] numpy-base                1.19.2           py38hfa32c7d_0    defaults
[conda] pytorch                   1.7.0           py3.8_cuda11.0.221_cudnn8.0.3_0    pytorch
[conda] torchaudio                0.7.0                      py38    pytorch
[conda] torchaudio-contrib        0.1                      pypi_0    pypi
[conda] torchcontrib              0.0.2                    pypi_0    pypi
[conda] torchvision               0.8.1                py38_cu110    pytorch

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100040

Fixes #51334

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -723,6 +724,29 @@ def atan(g: jit_utils.GraphContext, self):
     return g.op("Atan", self)
 
 
+@_onnx_symbolic("aten::atan2")
+@_beartype.beartype
+def atan2(g: jit_utils.GraphContext, self, other):
+    # self is y, and other is x on coordinate
+    slope = g.op("Div", self, other)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:731]: what about 0s in other?

Comment (titaiwangms) [reply torch/onnx/symbolic_opset9.py:731]: That should be included as arctangent(inf) is valid. I guess the case is included in OP_DB, but not sure.

Comment (titaiwangms) [reply torch/onnx/symbolic_opset9.py:731]: Both torch/numpy gives inf, so if ONNX gives Nan for example, I think that should be corrected.

Comment (BowenBao) [reply torch/onnx/symbolic_opset9.py:731]: > However, is 0 in denominator beyond ONNX spec?

Not sure, I suppose it should be inline with torch.

**Review by BowenBao — APPROVED** [review]

**Comment by justinchuby** [comment]
It occurred to me that we should be able to check the input type and conditionally cast integers to float32 so as to support all inputs.

**Comment by BowenBao** [comment]
Can it be? I haven't put much thought into it, wasn't sure if this holds true `int(atan2(float(args))) == atan2(args), when dtype(args) is int`

**Pull request closed without merging**